src/main/groovy/com/netgrif/application/engine/petrinet/domain/dataset/logic/action/ActionDelegate.groovy

```diff
@@ -227,112 +226,164 @@ class ActionDelegate {
+     *     make text, visible on transition when { condition.value == true }
+     * </pre>
+     * This code will change the field <i>text</i> behaviour to <i>visible</i> on given transition when field's <i>condition</i> value is equal to <i>true</i>.
      *
      * Example 2:
      * <pre>
-     *      text: f.textId;
+     *     condition: f.conditionId,
+     *     text: f.textId,
+     *     transition: t.transitionId,
+     *     anotherTransition: t.anotherTransitionId;
      *
-     *      if (condition)
-     *          make text, visible on each
+     *     make text, visible on ([transition, anotherTransition]) when { condition.value == true }
      * </pre>
-     * This code will change the field <i>text</i> behaviour to <i>visible</i> on each transition that contains the field <i>text</i>.
+     * This code will change the field <i>text</i> behaviour to <i>visible</i> on given transitions when field's <i>condition</i> value is equal to <i>true</i>.
+     *
+     * Example 3:
+     * <pre>
+     *     condition: f.conditionId,
+     *     text: f.textId;
+     *
+     *     make text, visible on transitions when { condition.value == true }
+     * </pre>
+     * This code will change the field <i>text</i> behaviour to <i>visible</i> on each transition that contains the field <i>text</i> when field's <i>condition</i> value is equal to <i>true</i>.
      * @param field which behaviour will be changed
      * @param behavior one of initial, visible, editable, required, optional, hidden, forbidden
      */
     def make(Field field, Closure behavior) {
-        [on: { Object optionalTransition ->
-            if (optionalTransition instanceof Transition) {
-                [when: { Closure condition ->
-                    if (condition()) {
-                        behavior(field, optionalTransition)
-                        saveFieldBehavior(field, optionalTransition)
-                    }
-                }]
-            } else if (optionalTransition instanceof Closure) {
-                if (optionalTransition == each) {
-                    useCase.petriNet.transitions.each {transitionEntry ->
-                        Transition trans = transitionEntry.value
-                        if (trans.dataSet.containsKey(field.stringId)) {
-                            behavior(field, trans)
-                            saveFieldBehavior(field, trans)
+        def behaviorClosureResult
+
+        [on: { Object transitionObject ->
+            [when: { Closure condition ->
+                if (condition()) {
+                    if (transitionObject instanceof Transition) {
+                        behaviorClosureResult = behavior(field, transitionObject)
+                        saveFieldBehavior(field, transitionObject, (behaviorClosureResult instanceof Set<FieldBehavior>) ? behaviorClosureResult : null)
+                    } else if (transitionObject instanceof List<Transition>) {
```

Comment:
`instanceof List<Transition>` is effectively a `instanceof List<?>` check because of runtime type erasure. It might be meaningful to check type of each object in the list to prevent hard-to-debug runtime errors.

Reply:
I added suggested implementation. Each element will be checked and in case of incorrect type, error is thrown.